clap_complete/src/dynamic/completer.rs

```diff
@@ -194,15 +194,7 @@ fn complete_arg(
                             completions.extend(
                                 complete_arg_value(value.to_str().ok_or(value), arg, current_dir)
                                     .into_iter()
-                                    .map(|comp| {
-                                        CompletionCandidate::new(format!(
-                                            "--{}={}",
-                                            flag,
-                                            comp.get_content().to_string_lossy()
-                                        ))
-                                        .help(comp.get_help().cloned())
-                                        .visible(comp.is_visible())
-                                    }),
+                                    .map(|comp| comp.add_prefix(format!("--{}=", flag))),
```

Comment:
Please put these refactor commits *before* the feature work.  Part of the intent is to clean up the feature work commit so we can better see whats happening in reviewing it.